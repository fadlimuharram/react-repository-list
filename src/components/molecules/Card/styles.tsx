import styled from 'styled-components';

export const Wrapper = styled.div`
	display: flex;
	align-items: center;
	flex-direction: column;
	width: 400px;
	height: 400px;
	background-color: #fff;
	box-shadow: 0 1rem 2rem rgb(0 0 0 / 15%);
	margin-top: 8rem;
	margin-right: 1rem;
	margin-left: 1rem;
	transition: all 0.5s;
	word-break: break-word;
	text-align: center;
	&:hover {
		transform: scale(105%);
	}

	ul {
		list-style: none;
		margin: 0;
		padding: 0;
		margin-top: 1rem;
		li {
			text-align: center;

			&:not(:last-child) {
				border-bottom: 1px solid #eee;
			}
		}
	}
`;
