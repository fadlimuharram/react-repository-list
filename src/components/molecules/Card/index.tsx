import React from 'react';
import Heading1 from '../../atoms/label/Heading1';
import SubHeading from '../../atoms/label/SubHeading';
import { Wrapper } from './styles';

interface IProps {
	owner: string;
	name: string;
	url: string;
	language: string;
	defaultBranch: string;
}

const Card: React.FC<IProps> = ({ name, url, language, owner, defaultBranch }) => {
	return (
		<Wrapper data-testid="card">
			<Heading1>{name}</Heading1>
			<ul>
				<li>
					<SubHeading>{owner}</SubHeading>
				</li>
				{language && (
					<li>
						<SubHeading>{language}</SubHeading>
					</li>
				)}

				<li>
					<SubHeading>{defaultBranch}</SubHeading>
				</li>
				<li>
					<a href={url} target="__blank">
						<SubHeading>Link</SubHeading>
					</a>
				</li>
			</ul>
		</Wrapper>
	);
};

export default Card;
