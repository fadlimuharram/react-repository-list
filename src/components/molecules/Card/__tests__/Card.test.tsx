import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Card from '..';

afterEach(() => {
	cleanup();
});

const data = {
	owner: {
		login: 'fadlimuharram',
	},
	name: 'AbsensiApp',
	url: 'https://api.github.com/repos/fadlimuharram/AbsensiApp',
	language: 'Javascript',
	defaultBranch: 'Master',
};

test('should render First Card', () => {
	render(
		<Card
			owner={data.owner.login}
			name={data.name}
			url={data.url}
			language={data.language}
			defaultBranch={data.defaultBranch}
		/>,
	);
	const elm = screen.getByTestId('card');
	expect(elm).toBeInTheDocument();
	expect(elm).toHaveTextContent(data.owner.login);
	expect(elm).toHaveTextContent(data.name);
	expect(elm).toHaveTextContent(data.language);
	expect(elm).toHaveTextContent(data.defaultBranch);
});

test('Match Card Snapshot', () => {
	const tree = renderer
		.create(
			<Card
				owner={data.owner.login}
				name={data.name}
				url={data.url}
				language={data.language}
				defaultBranch={data.defaultBranch}
			/>,
		)
		.toJSON();

	expect(tree).toMatchSnapshot();
});
