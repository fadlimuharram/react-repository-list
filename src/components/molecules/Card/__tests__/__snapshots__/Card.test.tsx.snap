// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Card Snapshot 1`] = `
<div
  className="sc-dlnjwi ilWMHm"
  data-testid="card"
>
  <h1
    className="sc-bdnxRM fxAJjl"
    data-testid="heading-1"
  >
    AbsensiApp
  </h1>
  <ul>
    <li>
      <h1
        className="sc-gtsrHT fOnSCr"
        data-testid="sub-heading"
      >
        fadlimuharram
      </h1>
    </li>
    <li>
      <h1
        className="sc-gtsrHT fOnSCr"
        data-testid="sub-heading"
      >
        Javascript
      </h1>
    </li>
    <li>
      <h1
        className="sc-gtsrHT fOnSCr"
        data-testid="sub-heading"
      >
        Master
      </h1>
    </li>
    <li>
      <a
        href="https://api.github.com/repos/fadlimuharram/AbsensiApp"
        target="__blank"
      >
        <h1
          className="sc-gtsrHT fOnSCr"
          data-testid="sub-heading"
        >
          Link
        </h1>
      </a>
    </li>
  </ul>
</div>
`;
